opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -65,6 +66,19 @@ impl Resource {
         }
     }
 
+    /// Insert and entry if it does not exist or is an empty string
+    fn insert(&mut self, key: Key, value: Value) {
+        match self.attrs.entry(key) {
+            Entry::Occupied(mut e) if e.get().to_string().is_empty() => {
```

Comment:
nit perf improvement:

```suggestion
            Entry::Occupied(mut e) if e.get().as_str().is_empty() => {
```